<a href="https://colab.research.google.com/github/srilakshmi-saladi/bdaassignment/blob/main/srilakshmimanikyamba_saladi_hw1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Python Review Assignment
16 points

Due by midnight September 11, 2025


**To avoid erasing the sample output, please insert a new cell above the sample output**. Write your script inside the new cell and run. Verify your output with the sample output

For lab submission, please rename your file using the following format

`firstname_familyname_hw1.ipynb`

### please enter your name and 700# in this cell

700769896




### Task 1
Define a string `s1` with content 'Good morning John'. Count the number of characters in `s1`

In [2]:
s1 = 'Good morning John'


In [3]:
len(s1)

17

17

### Task 2
Check if the word 'morning' is contained in `s1`

In [4]:
'morning' in s1

True

True

### Task 3
Use a function to replace 'John' with 'David' in `s1`. Show the content of the new string

In [10]:
def replace_name(s1):
    return s1.replace('John','David')

new_s1 = replace_name(s1)
new_s1

'Good morning David'

'Good morning David'

### Task 4
Use your UCM email address to create a string s2. Extract the string `ucmo.edu`

In [12]:
s2 = 'sxv78190@ucmo.edu'
s2.split('@')[1]

'ucmo.edu'

'ucmo.edu'

### Task 5
Create a list x with integers from 1 to 30. Create a new list y which only conatins the odd numbers from list x. You are not allowed to mannually type out the lists

In [17]:
x = [i for i in range(1,31)]
x

[1,
 2,
 3,
 4,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 20,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28,
 29,
 30]

In [18]:
y = [i for i in x if i%2==1]
y

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29]

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29]

### Task 6
Create a new list z with each element is a squre of the corresponding element from list y. You are not allowed to mannually type out the list

In [19]:
z = [i*i for i in y]
z

[1, 9, 25, 49, 81, 121, 169, 225, 289, 361, 441, 529, 625, 729, 841]

[1, 9, 25, 49, 81, 121, 169, 225, 289, 361, 441, 529, 625, 729, 841]

### Task 7
Extract `[729, 625, 529, 441, 361, 289, 225, 169]` from list z. You are not allowed to mannually type out the list

In [20]:
z[-2:-10:-1]

[729, 625, 529, 441, 361, 289, 225, 169]

[729, 625, 529, 441, 361, 289, 225, 169]

### Task 8
Reverse the order of z. You are not allowed to mannually type out the list

In [21]:
z[-1:-16:-1]

[841, 729, 625, 529, 441, 361, 289, 225, 169, 121, 81, 49, 25, 9, 1]

[841, 729, 625, 529, 441, 361, 289, 225, 169, 121, 81, 49, 25, 9, 1]

### Task 9
Given the following list, extract the string 'Get Me'

In [22]:
lst = [1, 2, [3, 4], [5, [100, 200, ['Get Me']], 300, 11], 1, 9]
lst[3][1][2][0]


'Get Me'

'Get Me'

### Task 10
Sort the following list of strings by their lengths. You are not allowed to mannually type out the list

In [26]:
x = ['ccc', 'dddd', 'bb', 'ggggggg', 'a', 'eeeee', 'ffffff']

x.sort()
x

['a', 'bb', 'ccc', 'dddd', 'eeeee', 'ffffff', 'ggggggg']

['a', 'bb', 'ccc', 'dddd', 'eeeee', 'ffffff', 'ggggggg']

### Task 11
Extract the string 'Get Me' from the following dictionary d

In [27]:
d = {'k1':[1, 2, 3, {'tricky':['oh', 'man', 'crazy', {'target':[1, 2, 3, 'Get Me']}]}]}
d['k1'][3]['tricky'][3]['target'][3]

'Get Me'

'Get Me'

### Task 12
Use lambda expressions and the `filter()` function to filter out words from the following list that don't start with the letter 's'

In [28]:
seq = ['soup', 'apple', 'salad', 'banana', 'spaghetti', 'spinach', 'kiwi']


In [29]:
list(filter(lambda s: not s.startswith('s'), seq))

['apple', 'banana', 'kiwi']

['apple', 'banana', 'kiwi']

### Task 13
Count the number of unique characters in the following string s. Please ignore the white space and character case. For example, 'P' and 'p' are considered as one character

In [30]:
s = 'Python programming is fun'
len(set(l.lower() for l in s if l!=' '))

14

14

### Task 14
Use lambda expressions and the `map()` function to count the length of each string in the list

In [31]:
seq = ['soup', 'apple', 'salad', 'banana', 'spaghetti', 'spinach', 'kiwi']
list(map(lambda s: len(s), seq))


[4, 5, 5, 6, 9, 7, 4]

[4, 5, 5, 6, 9, 7, 4]

### Task 15
Create a function that counts the number of times the word 'dog' occurs in a string. Ignore edge cases.

In [33]:


def countDog(s):
    return s.lower().split().count('dog')
countDog('This dog looks differently from that dog in the dog shelter!')

3

3

### Task 16
Write a `for` loop to process employee information listed in the cell below. For each employee with Level 1, increase the salary by 2000 and adjust the Level from 1 to 2. Display all employee information afterwards.

In [34]:
employee = {
    'Elon Musk':{'Department':'Sales', 'Salary':6000, 'Level':2},
    'Bill Gates': {'Department':'Engineering', 'Salary':10000, 'Level':3},
    'Donald Trump':{'Department':'Public Relations', 'Salary':3000, 'Level':1},
    'Joe Biden':{'Department':'Human Resources', 'Salary':5000, 'Level':2},
    'Mark Zuckerberg':{'Department':'Engineering', 'Salary':10000, 'Level':3},
    'Jensen Huang':{'Department':'Sales', 'Salary':4000, 'Level':1}
}

In [37]:
for name, info in employee.items():
    if info['Level'] == 1:
        info['Salary'] += 2000
        info['Level'] = 2
employee

{'Elon Musk': {'Department': 'Sales', 'Salary': 6000, 'Level': 2},
 'Bill Gates': {'Department': 'Engineering', 'Salary': 10000, 'Level': 3},
 'Donald Trump': {'Department': 'Public Relations',
  'Salary': 5000,
  'Level': 2},
 'Joe Biden': {'Department': 'Human Resources', 'Salary': 5000, 'Level': 2},
 'Mark Zuckerberg': {'Department': 'Engineering', 'Salary': 10000, 'Level': 3},
 'Jensen Huang': {'Department': 'Sales', 'Salary': 6000, 'Level': 2}}

{'Elon Musk': {'Department': 'Sales', 'Salary': 6000, 'Level': 2}, 'Bill Gates': {'Department': 'Engineering', 'Salary': 10000, 'Level': 3}, 'Donald Trump': {'Department': 'Public Relations', 'Salary': 5000, 'Level': 2}, 'Joe Biden': {'Department': 'Human Resources', 'Salary': 5000, 'Level': 2}, 'Mark Zuckerberg': {'Department': 'Engineering', 'Salary': 10000, 'Level': 3}, 'Jensen Huang': {'Department': 'Sales', 'Salary': 6000, 'Level': 2}}
